### DISCUSSION TITLE
Why i keep getting this error? Error on Release.php line 49

### DISCUSSION THREAD

**Discussion by knewitzgui** [deployphp/deployer#3467] [Help needed]
Every time i make a deploy i get this error and the deploy fail, to work i need to delete the folder "releases" and re-run the deploy, then i dont get this error and all goes well.
But the next time i try to run, this error show up and i need to delete "releases" again, and so on.

![image](https://user-images.githubusercontent.com/49166923/215297999-7a42c069-6c54-481d-bab7-d892460e8f01.png)
 

By the way im using the deployer action on github.

**Comment by antonmedv** [comment]
This is because array key 2 is undefined.

**Comment by knewitzgui** [comment]
Yeah, this is clear, what is not clear is why this happens even for the first deploy that i made.
Simply doesnt make sense.

**Reply by antonmedv** [reply]
If array is like `[1, 2]`, trying to access key 2 will provide an error:

```php
$array = [1, 2];

$value = $array[2]; // This is trigger the error.
```

To mitigate this issue you can try to use try-catch syntax:

```php
$array = [1, 2];

try {
    $value = $array[2];
} catch ($e) {
    echo "no key 2";
}
```

Here is more helpful link for you:

- https://www.php.net/manual/en/language.exceptions.php
- https://stackoverflow.com/questions/16675753/php-fastest-way-to-handle-undefined-array-key

**Comment by mohamedds-12** [comment]
Check you're releases log file for an empty line at the begining or the end, if so remove it, if empty, make sure the cursor is in the first line not the second.
